site-scrapers/Hannaford.js

```diff
@@ -23,48 +23,76 @@ module.exports = async function GetAvailableAppointments(browser) {
 
 async function ScrapeWebsiteData(browser) {
     const page = await browser.newPage();
-    await page.goto(sites.Hannaford.website);
-    await page.solveRecaptchas().then(({ solved }) => {
-        if (solved.length) {
-            return page.waitForNavigation();
-        } else {
-            return;
+    const results = [];
+    for (let [i, loc] of sites.Hannaford.locations.entries()) {
+        await page.goto(sites.Hannaford.website);
+        await page.solveRecaptchas().then(({ solved }) => {
```

Comment:
Did you see my comment about sending a request to https://hannafordsched.rxtouch.com/rbssched/program/covid19/Calendar/PatientCalendar?facilityId=52076&month=2&year=2021&snapCalendarToFirstAvailMonth=false (with params modified) for each site? I'd much prefer to do that since a) then we wouldn't have to pay someone to solve Captchas for us and b) we wouldn't be relying on the website design

Reply:
Congrats on your CNBC interview! Yes I've been trying to intercept the XHR but not able to get the response for some reason. On the captcha front, that code was already there. Right now no captchas are present but there may be some in the future.

Reply:
Also, that first response just has whether a day has availability. I presume you want the number of appointments right? If so, that would be a second call to `rbssched/program/covid19/Calendar/PatientCalendarDay` which returns the number of slots `{
    "Success": true,
    "Message": "Day successfully retrieved",
    "Data": {
        "Date": "2/18/2021",
        "Rows": [
            {
                "RowTitle": "9:15 am",
                "TimeSlotString": "2/18,9:15,am",
                "TimeSlotTime": {
                    "Ticks": 333000000000,
                    "Days": 0,
                    "Hours": 9,
                    "Milliseconds": 0,
                    "Minutes": 15,
                    "Seconds": 0,
                    "TotalDays": 0.38541666666666663,
                    "TotalHours": 9.25,
                    "TotalMilliseconds": 33300000,
                    "TotalMinutes": 555,
                    "TotalSeconds": 33300
                },
                "EventDescription": null
            }
        ],
        "IsEvent": false
    }
}`

Reply:
Ah. Yeah I don't have the results in front of me but if there's a better ajax call then yes please, by all means.
And my bad re: captchas. If it's just as easy to make the calls yourself instead of using Puppeteer that'd be great so we can get rid of the captcha solver anymore. But that'd be a bonus.